components/ui/AppLink.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <component
+    :is="isNuxtLink ? 'nuxt-link' : 'a'"
+    :href="isAnchor && url"
+    :to="isNuxtLink && url"
+    style="cursor:pointer"
+    :rel="isExternal && 'noopener'"
+    :target="isExternal && '_blank'"
+    @click="segment && $trackClickEvent(segment)"
+  >
+    <slot />
+  </component>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+@Component
+export default class extends Vue {
+  @Prop(String) url
+  @Prop(Object) segment
+  @Prop({ type: Boolean, default: false }) isStatic
+
+  isExternal: boolean = this.url.startsWith('http')
```

Comment:
Be careful, the value of `this` inside the class scope is the class itself and you need to refer to the component instance. Convert these to getters.